### DISCUSSION TITLE
Why implicitly[sourcecode.Enclosing] inside T.source fails to compile?

### DISCUSSION THREAD

**Discussion by jk-1** [com-lihaoyi/mill#1523] [Q&A] [answered]
I have following `build.sc` module

    object myScalaMod extends ScalaModule {

      override def scalaVersion: Target[String] = "2.13.6"

      override def ammoniteVersion: Target[String] = "2.4.0"

      def fooSources: Source = T.source {
        val enclosing = implicitly[sourcecode.Enclosing]
        println(s"enclosing: ${enclosing}")
        os.pwd / 'src
      }
    }

When trying to run it:

    mill show myScalaMod.fooSources
    Compiling /home/jk/workspace.exp/hands-on-scala/mill01/build.sc
    Error while emitting build.sc
    value enclosing
    Compilation Failed

If I comment out the line `println(s"enclosing: ${enclosing}")` the file compiles fine.

This error happens only with `T.source,` not with `T`,  `T.task`, `T.input`, `T.worker`, `T.persistent` or plain old scala function.

Is this bug or why is `T.source` behaving differently from others?

Is it possible to get more detailed error message on this? At least line number would be very valuable information.

**Comment by lefou** [comment]
Thanks for reporting this issue. I think this is an oversight in the implementation. I'm creating an issue from this discussion.

**Comment by lefou** [comment, marked as answer]
I create issue #1541.

**Reply by lefou** [reply]
#1541 was merged and the issue should be fixed.
